File: JavaScript/Assignment.js
class MyString {

    constructor(string) {
        string == null ? this.str = "Hello, World!" : this.str = string;      
        this.letters = {};
    }
    
    frequency() {
        this.letters = {};
        let string = this.str.toLowerCase().replace(/\W/g, '');            
        for (let i = 0; i < string.length; i++) {
            let x = string[i];
            this.letters[x] = (this.letters[x] || 0) + 1;
        }
    }
    
    histogram() {
        this.frequency();
        let start = "a", end = "z";
        for (let i = start.charCodeAt(0); i <= end.charCodeAt(0); i++) {
            let output = String.fromCharCode(i) + ": ";
            let x = String.fromCharCode(i);
            for (let j = 0; j < this.letters[x]; j++) {
                output += "*";
            }
            console.log(output);
        }
    }
}

File: Ruby/Assignment.rb
class MyString
    attr_accessor :str
    attr_reader :letters
    
    def initialize(string="Hello, World!")
        @str = string
        @letters = {}
    end

    def histogram
        frequency
        ("A".."Z").each {|x| puts "#{x}: " + "*" * letters[x]}
    end

    private
    
    attr_writer :letters

    def frequency
        freq = String.new(str)
        freq.upcase!.gsub!(/\W+/, '')
        letters.clear
        letters.default = 0
        freq.each_char {|char| letters[char] += 1}
    end
end